embeddings/task/lightning_task/text_classification.py

```diff
@@ -0,0 +1,93 @@
+from typing import Any, Dict, Optional, Tuple, Type
+
+import numpy as np
+import torch
+from pytorch_lightning.utilities.types import STEP_OUTPUT
+from torch.utils.data import DataLoader
+from torchmetrics import F1, Accuracy, MetricCollection, Precision, Recall
+from transformers import AutoModel, AutoModelForSequenceClassification
+
+from embeddings.data.datamodule import HuggingFaceDataset
+from embeddings.task.lightning_task.lightning_task import HuggingFaceLightningTask
+
+
+class TextClassificationTask(HuggingFaceLightningTask):
+    def __init__(
+        self,
+        num_labels: int,
+        model_name_or_path: str,
+        downstream_model_type: Type["AutoModel"] = AutoModelForSequenceClassification,
```

Comment:
What is the use of this argument? Could we pass own model class here?

Reply:
for now the text_classification datamodule works only with predefined automodel classes from HF; you could pass your own model if it inherits from AutoModel and has a specific return type (e.g. TokenClassifierOutput in this case)

Reply:
is there any other model different than `AutoModelForSequenceClassification` for text classification right now? Maybe it would be worth to consider if we wrap this model by own interface. That would speedup the time for building custom model classes rather than relying on HF interfaces which may be subject of change.

Reply:
the previous PR (torch-pipelines) allowed for specifying your own models and they inherited from the task and implemented the forward method; this was simplified and allows for hf automodels but user could write his own automodel (I haven't looked how it should be implemented tho) 
however I wanted to move this to the one of the next PRs since we are using the automodels now anyway and we could write this functionalities together with poolings etc.

Reply:
So maybe as for now I would remove this parameter. And handle it in other PR's.

Reply:
done